linux_basics/README.md

```diff
@@ -0,0 +1,194 @@
+
+# Piping
+
+Piping is a form of redirection. It can be used to route output in any direction/location you want.
+
+Here's an example:
+
+Let's say your on the Desktop and you have three directories (a, b,c, apple, carrot, christopherColumbus). 
+
+To see a list of those directories, you'd just type `ls`. 
+
+***Output:***
+
+a
+
+b
+
+c
+
+apple
+
+carrot
+
+christopherColumbus
+
+
+Now, if you wanted to search for directories that had the letter 'a' in them, you'd use the grep command (grep = search). 
+
+To list out all the directories with ONLY a, type: `ls | grep a`
+
+The ls stands for list, and the | acts as a pipe. We're directing the output to only feed through directories/documents that contain the letter 'a'. 
+
+OUTPUT: 
+
+a
+
+apple
+
+carrot
+
+Note, `ls grep a` does not work. Ask yourself this if you don't understand why: what are you trying to execute? ls, grep, or a? How is the computer supposed to know what  you're talking about? The `|` allows you to "funnel" the output and get only what you want. 
+
+Another note, you can have multiple `|` in an expression. 
+
+Ex: 
+
+`ls | grep a | grep c`
+
+Start with `grep c`. This will list only the documents/directories that contain the letter c.
+
+The `grep a` then funnels the output further; only output that contains the letter a will be listed. 
+
+***OUTPUT:***
+
+carrot
+
+***Note, christopherColumbus and 'c' will not be listed because they do not contain the letter a - thus the term, piping/funneling***
+
+
+# Redirecting
+
+Redirecting relies on the following two symbols: `<` & `>`. They are fairly simple to use if you think about the acute angled side as the funnel. 
+
+Example: 
+
+INPUT > FUNNELED OUTPUT
+
+Practical Example:
+
+Let's say you want to print the contents of one file to another. This can be accomplished with redirecting. 
+
+  `touch file file2` #touch creates a file without opening a text editor. In this case, we've created two files, file & file2.
+  
+  `nano file` #open the text editor for file, write anything you want. In this case, we'll just write `Hello World`. 
+  
+  Now to transfer the contents of file to file2, we'll "redirect" the output of the `cat` command.
+  
+  `cat file > file2` Now the contents of file2 have been replaced with the contents of file. If you wish to append rather than replace, use >> instead.
+  
+  `cat file >> file2` 
+  
+  To check that the transfer was successful, type `cat file2`. The output should now match file.
+  
+  You've made it to the end of this brief introduction of Linux. If you understand the above steps, you have an excellent foundation. With practice, you'll be able to get around a terminal with no issue at all. 
+  
+  Recommended next steps: 
+    
+    1). PRACTICE PRACTICE PRACTICE!
```

Comment:
very detailed! thanks a lot, one thing before you merge can you rename the root folder (linux_basics) to LinuxBasics just so it matches the format of the other folders.

thank you for doing this, its detailed and will be helpful down the line :))